Develop

AlexandrZhezheria opened the pull request:
![1](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/ab9b3ca6-a021-40b6-b5af-9cbbfc09eec9)
![Screenshot_1](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/7944864c-880a-4d6d-b784-61ce24f1352b)
![Screenshot_2](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/e0426b37-c298-4056-8137-45805471a5b9)
![Screenshot_3](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/4012a9e2-7904-4d9a-a22d-78994cbe923a)
![Screenshot_4](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/15f7f96c-04e9-480a-b9df-edaecbc19fdc)
![Screenshot_5](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/4c302e55-0de4-4dab-bd9d-47fa082c2f89)
![Screenshot_6](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/000d4580-2749-4ffa-8241-dda1ae6984df)
![Screenshot_7](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/b78933f4-792d-4fc4-8378-2a5129b0e39e)
![Screenshot_8](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/02ebff63-c608-41a1-9885-4cbe6a28ceff)
![Screenshot_9](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/0164afbc-5330-45fa-a72c-831b8723b32b)
![Screenshot_10](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/dd2e036e-1f76-488d-b520-2271bd3a36a5)
![Screenshot_11](https://github.com/AlexandrZhezheria/Restaurant-Kitchen-Service/assets/108021795/4d7652b0-8f3b-41ad-b8b1-7540ccba9353)

AlexandrZhezheria committed:
Create admin superuser

AlexandrZhezheria committed:
Create APP restaurant_service

AlexandrZhezheria committed:
install debug_toolbar
add requirements.txt

AlexandrZhezheria committed:
install crispy_bootstrap5

AlexandrZhezheria committed:
add static  folder
add stile.css in static  folder

AlexandrZhezheria committed:
add models to models.py

AlexandrZhezheria committed:
add register to admin.py
add path to urls.py
add classes to views.py
create forms.py
add forms in forms.py

AlexandrZhezheria committed:
add cooks_list.html,  cook_form.html, cook_detail.html, cook_confirm_delete.html

AlexandrZhezheria committed:
add dish_list.html,  dish_form.html, dish_detail.html, dish_confirm_delete.html, dish_type_list.html, dish_type_form.html, dish_confirm_delete.html

AlexandrZhezheria committed:
Delete DB
Create DB and make migrations
Update cook_detail.html, dish_detail.html, forms.py, models.py, urls.py, sidebar.html

AlexandrZhezheria committed:
rename dishtype_confirm_delete.html, dishtype_form.html
refactoring urls.py

AlexandrZhezheria committed:
added a base-presentations.html from the repository for editing under its task
copy directory asserts rom the repository for editing under its task

AlexandrZhezheria committed:
add in settings.py assets root
add context_processors.py
add includes footer.html, navigation.html, scripts.html

AlexandrZhezheria committed:
add index.html and update for your task

AlexandrZhezheria committed:
update footer.html, index.html, navigation.html for your task

AlexandrZhezheria committed:
update cook_list.html for your task

AlexandrZhezheria committed:
update cook file html and dish file html for your task
update search_form.html
add img for my project

AlexandrZhezheria committed:
update dish type file html for your task
add img for my project

AlexandrZhezheria committed:
update pagination.html

AlexandrZhezheria committed:
update login.html

AlexandrZhezheria committed:
rolling back to working version

AlexandrZhezheria committed:
update navigation.html

AlexandrZhezheria committed:
delete unnecessary img

Grayder0152 commented on restaurant/settings.py:
```diff
@@ -15,7 +15,6 @@
-
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
 
```
Add `SECRET_KEY` variable to env

Grayder0152 commented on restaurant_service/forms.py:
```diff
@@ -0,0 +1,72 @@
+        return validate_years_of_experience(self.cleaned_data["years_of_experience"])
+
+
+def validate_years_of_experience(
+        years_of_experience,
+):  # regex validation is also possible here
+    if len(years_of_experience) == 0:
+        raise ValidationError("years of experience cannot be 0")
+
+    return years_of_experience
```
move this function before classes

Grayder0152 commented on restaurant_service/forms.py:
```diff
@@ -0,0 +1,72 @@
+
+def validate_years_of_experience(
+        years_of_experience,
+):  # regex validation is also possible here
```
add typing

Grayder0152 commented on db.sqlite3:
add this file to gitignore

Grayder0152 commented on restaurant_service/models.py:
```diff
@@ -0,0 +1,43 @@
+        max_length=1000,
+        help_text="Select a description for this dish..."
+    )
+    price = models.DecimalField(max_digits=7, decimal_places=2, default=0.00)  # Добавлено значение по умолчанию
```
remove comment

Grayder0152 replied:
```suggestion
    price = models.DecimalField(max_digits=7, decimal_places=2, default=0.0)  # Добавлено значение по умолчанию
```

AlexandrZhezheria committed:
add db.sqlite3 to ,gitignore

AlexandrZhezheria committed:
create .env, add .env to .gitignore
add python-dotenv~=1.0.0 to requirements.txt
Add SECRET_KEY variable to env

AlexandrZhezheria committed:
update default=0.00 to default=0.0 and remove comment

AlexandrZhezheria committed:
added typing annotations to the function parameters

AlexandrZhezheria committed:
update paths STATIC_ROOT, STATICFILES_DIRS, DATABASES
validate_years_of_experience function is defined before the classes that use it.

AlexandrZhezheria committed:
added newline for restaurant_service/templates/restaurant_service/dishtype_form.html
restaurant_service/templates/restaurant_service/dish_confirm_delete.html
restaurant_service/templates/includes/head.html
restaurant_service/templates/includes/navigation-auth.html

AlexandrZhezheria committed:
I`m rewrite def toggle_assign_to_dish like a class with post method in views.py

AlexandrZhezheria committed:
I`m rewrite def toggle_assign_to_dish like a class with post method in views.py

AlexandrZhezheria committed:
remove last line from footer.html

AlexandrZhezheria committed:
add new line from navigation.html

AlexandrZhezheria committed:
update .gitignore

AlexandrZhezheria committed:
update urls.py

AlexandrZhezheria committed:
Update views.py , to be able to sort Cooks by ID

AlexandrZhezheria committed:
Make README.md looks betters

AlexandrZhezheria committed:
Update settings.py

AlexandrZhezheria committed:
Update settings.py, views.py, requirements.txt

AlexandrZhezheria committed:
staticfiles managing

AlexandrZhezheria committed:
added build.sh script for deployment

AlexandrZhezheria committed:
added build.sh script for deployment

AlexandrZhezheria committed:
upgrade pywin32==305 to pywin32==306

AlexandrZhezheria committed:
upgrade pywin32==306 to pywin32

AlexandrZhezheria committed:
upgrade requirements.txt

AlexandrZhezheria committed:
upgrade requirements.txt

AlexandrZhezheria committed:
upgrade requirements.txt

AlexandrZhezheria committed:
upgrade requirements.txt

AlexandrZhezheria committed:
add ALLOWED_HOSTS "restaurant-kitchen-service-xi2k.onrender.com"

AlexandrZhezheria committed:
add link to deployed application

AlexandrZhezheria committed:
added test user
